### PR TITLE
misc: Use Button component from Cypress Design System

### PR DESCRIPTION
**Pull request by mike-plummer**
- Closes cypress-io/cypress-internal#589

### Additional details
Switching some of the Button elements in the app to use `@cypress-design/vue-button`

There are multiple uses that cannot be converted yet due to bugs identified in prop reactivity (unable to disable then enable a button, for example) - fixes are being worked in the DS repo.

Most buttons should be nearly identical, but there are some changes to border/hocus color - certain button uses do not have a corresponding variant in the design system: [Slack](https://cypressio.slack.com/archives/C3WD9SGKW/p1684164816054609)

### Steps to test
Basic smoke tests if desired, existing component tests & Percy should cover most cases

### How has the user experience changed?
There should not be any real user impact - small changes to style that will likely go unnoticed

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by astone123** [comment]
@mike-plummer I left some comments in the Percy diff. Looks like some button icons changed appearance, not sure if that's acceptable for this or not

**Comment by mike-plummer** [comment]
> @mike-plummer I left some comments in the Percy diff. Looks like some button icons changed appearance, not sure if that's acceptable for this or not

@astone123 Good 👀 . I'm fixing one icon that I broke on the `RequestAccessButton`. The remainder of items you pointed out are either changes to the icon itself from the Design System (fill colors on the "User" icon) or aligning icon coloration with the design system (indigo color within button) which I believe were part of the intent of this ticket (align with design system). I'll defer to product (@marktnoonan ?) to confirm or correct

File: packages/app/cypress/e2e/specs_list_latest_runs.cy.ts [hunk 1]
```diff
@@ -194,7 +194,7 @@ describe('App/Cloud Integration - Latest runs and Average duration', { viewportW
       cy.get('.v-popper__popper--shown')
       .should('contain', 'Connect to Cypress Cloud to see the status of your latest runs')
       .find('button')
-      .should('have.text', 'Log in to Cypress Cloud')
+      .should('contain.text', 'Log in to Cypress Cloud')
```
Comment (warrensplayer) [inline packages/app/cypress/e2e/specs_list_latest_runs.cy.ts:197]: Why did you have to change these assertions from 'have.text' to 'contain.text'?

**Review by warrensplayer — COMMENTED** [review]
Two suggestions for moving forward with bringing in more Design System components:

1) We need a way to document the patterns of usage within the App codebase.  We have a version of that now with the existing components in `frontend-shared` with their component tests.   This could be with documentation and/or with component tests just to exercise the design system components the way the App needs to use them.  This should also call out when they should not be used yet (e.g. the button is not reactive yet)
2) For patterns of component usage that do not exist in the design system yet, one option instead of waiting to update the design system is to write wrapper components in the App code to use.  These could be replaced if/when the design system is updated.  For example, the main repeating pattern I see in these changes is with a button with an icon.  Additional utility classes after to be added to space out the icon.

Open to other ideas!

File: packages/app/cypress/e2e/specs_list_latest_runs.cy.ts (hunk 1 above)
Comment (mike-plummer) [reply packages/app/cypress/e2e/specs_list_latest_runs.cy.ts:197]: `SpecHeaderCloudDataTooltip` was updated to use the DS version of Button which relies on passing prefix/suffix icons as part of the default slot. New content looks like this:

```
>
  <IconUserGeneralOutline />
  {{ t('specPage.cloudLoginButton') }}
</Button>
```

which treats the newline as a space when determining `text` content. I think pushing the slot content onto one line would resolve the issue but is pretty ugly and I'm not sure whether the linter would he happy either.

**Comment by mike-plummer** [comment]
@warrensplayer I think the "ExternalLink" and "Router" integration wrapper(s) belong in app, not in the design system. It was just wasn't feasible given the one-day bound on this ticket. Long-term all buttons *should* directly use the DS button *or* the wrapper if they use `href` or `to` props, and the wrapper is purely a data integration component (no styling, behavior, etc).

I agree that the "Icon with `gap-[8px]` pattern is painful. The existing `Button` nicely handled prefix/suffix icon spacing for us, but I believe design is trying to treat buttons with icons as a "non-default" scenario in the DS

**Comment by lmiller1990** [comment]
I moved this to draft for now - it's not clear to me whether I should be reviewing this or not, it's currently in blocked.

Once we decide if more work is needed, we can move it back to in review.

**Comment by ryanjwilke** [comment]
@mike-plummer @lmiller1990 Let me or @elevatebart know if there's anything either of us can do to help get this work over the finish line. It would probably be good to make sure we're using the latest version of the DS components wherever possible.

**Comment by jennifer-shehane** [comment]
Closing as stale

**Pull request closed without merging**